pkg/providers/instancetype/instancetype.go

```diff
@@ -160,12 +167,17 @@ func (p *DefaultProvider) List(ctx context.Context, kc *v1.KubeletConfiguration,
 			instanceTypeLabel: *i.InstanceType,
 		}).Set(float64(aws.Int64Value(i.MemoryInfo.SizeInMiB) * 1024 * 1024))
 
+		var vmMemoryOverhead int64
+		if cached, ok := p.vmMemoryOverheadCache.Get(fmt.Sprintf("%s-%d", *i.InstanceType, amiHash)); ok {
```

Comment:
We probably want to keep track of the other detail in the nodeclass, specifically from the kubelet

Reply:
Specifically here: https://github.com/kubernetes-sigs/karpenter/blob/887761eb533674aa29cb86e84b243781b7b0bf03/pkg/cloudprovider/types.go#L97-L99